runtime/mainnet/src/genesis.rs

```diff
@@ -0,0 +1,610 @@
+use parachains_common::{AccountId, AuraId, Balance};
+use pop_runtime_common::genesis::*;
+use sp_core::crypto::Ss58Codec;
+use sp_genesis_builder::PresetId;
+use sp_runtime::traits::AccountIdConversion;
+
+use crate::{
+	config::{
+		collation::PotId,
+		governance::SudoAddress,
+		monetary::{ExistentialDeposit, MaintenanceAccount, TreasuryAccount},
+	},
+	AssetsConfig, BalancesConfig, CouncilConfig, Runtime, SessionKeys, EXISTENTIAL_DEPOSIT, UNIT,
+};
+
+/// A development chain running on a single node, using the `mainnet` runtime.
+pub const MAINNET_DEV: &str = "pop-dev";
+/// Configures a local chain running on multiple nodes for testing purposes, using the `mainnet`
+/// runtime.
+pub const MAINNET_LOCAL: &str = "pop-local";
+/// A live chain running on multiple nodes, using the `mainnet` runtime.
+pub const MAINNET: &str = "pop";
+/// The available genesis config presets;
+const PRESETS: [&str; 3] = [MAINNET_DEV, MAINNET_LOCAL, MAINNET];
+
+/// The parachain identifier to set in genesis config.
+pub const PARA_ID: ParaId = ParaId::new(3395);
+
+/// Initial balance for genesis endowed accounts.
+const ENDOWMENT: Balance = 10_000_000 * UNIT;
+
+/// The default XCM version to set in genesis config.
+const SAFE_XCM_VERSION: u32 = xcm::prelude::XCM_VERSION;
+
+/// Struct used to declare assets that will be included at genesis.
+struct GenesisAsset {
+	id: u32,
+	owner: AccountId,
+	is_sufficient: bool,
+	min_balance: Balance,
+	name: Vec<u8>,
+	symbol: Vec<u8>,
+	decimals: u8,
+}
+
+/// Returns a JSON blob representation of the built-in `RuntimeGenesisConfig` identified by `id`.
+pub(crate) fn get_preset(id: &PresetId) -> Option<Vec<u8>> {
+	let patch = match id.as_str() {
+		MAINNET_DEV => development_config(),
+		MAINNET_LOCAL => local_config(),
+		MAINNET => live_config(),
+		_ => return None,
+	};
+	Some(
+		to_string(&patch)
+			.expect("serialization to json is expected to work. qed.")
+			.into_bytes(),
+	)
+}
+
+/// Returns a list of identifiers for available builtin `RuntimeGenesisConfig` presets.
+pub(crate) fn presets() -> Vec<PresetId> {
+	PRESETS.map(PresetId::from).to_vec()
+}
+
+/// Configures a development chain running on a single node, using the `mainnet` runtime.
+fn development_config() -> Value {
+	let mut endowed_accounts = dev_accounts();
+	endowed_accounts.push(MaintenanceAccount::get());
+	endowed_accounts.push(PotId::get().into_account_truncating());
+	endowed_accounts.push(TreasuryAccount::get());
+
+	genesis(
+		// Initial collators.
+		Vec::from([
+			// Single collator for development chain
+			(Keyring::Alice.to_account_id(), Keyring::Alice.public().into()),
+		]),
+		endowed_accounts,
+		Keyring::Alice.to_account_id(),
+		PARA_ID,
+		// AssetId reserved for DOT from AH.
+		vec![GenesisAsset {
+			id: 0,
+			owner: Keyring::Alice.to_account_id(),
+			is_sufficient: false,
+			min_balance: ExistentialDeposit::get(),
+			name: "DOT".into(),
+			symbol: "DOT".into(),
+			decimals: 10,
+		}],
```

Comment:
I realize you are getting conflicting code direction. However, for mainnet I would like to exclude this. 

I am still not sold that we should have devs use the fungibles API for DOT. It introduces complexity due to pallet-balances missing functionality that pallet-assets have.

By including, we can forget its there. By excluding, we have to consciously add it in -- when we confirm it works.  

If we want to reserve asset id 0, I recommend using another name than "DOT" as a placeholder.

Reply:
Happy to remove from mainnet.